Issue500 update data manager

EttoreZ opened the pull request:
This PR addresses #500, a warning was added to the `data/data_manager.py` script if any of the weather variables present in `data/categories.json` is missing from `wrapped.fum/resources/weather.csv`.

dhblum reviewed (changes requested):
@EttoreZ I think this functionality should go in the method ``save_data_and_jsons`` since that is invoked when the csvs are moved to the FMU when compiling the test case and is when other similar warnings are shown to a test case developer if there are issues with the different files.  

``load_data_and_jsons`` is called when the test case docker container is deployed and the warning message will be lost among alot of other container deployment messages, not known to a typical boptest user (who is not a test case developer), and may not be displayed at all if the container is deployed in the background or a remote server (i.e. on aws).

EttoreZ commented:
@dhblum thank you for the comments, they all make sense. 

- I moved the warning to the method `_append_csv_data` within `save_data_and_jsons `, which is where the .csv files are checked.
- I updated the release notes according to the suggestion.
